api/trace/span_context_test.go

```diff
@@ -106,6 +107,144 @@ func TestIsValidFromHex(t *testing.T) {
 	}
 }
 
+func TestIsValidSpanIDFromHex(t *testing.T) {
+	for _, testcase := range []struct {
+		name  string
+		sid   trace.SpanID
+		hex   string
+		valid bool
+	}{
+		{
+			name:  "Valid SpanID",
+			sid:   trace.SpanID([8]byte{128, 241, 152, 238, 86, 52, 59, 168}),
+			hex:   "80f198ee56343ba8",
+			valid: true,
+		}, {
+			name:  "Invalid SpanID with invalid length",
+			hex:   "80f198ee56343ba",
+			valid: false,
+		}, {
+			name:  "Invalid SpanID with invalid char",
+			hex:   "80f198ee56343bg8",
+			valid: false,
+		}, {
+			name:  "Invalid SpanID with uppercase",
+			hex:   "80f198ee56343bA8",
+			valid: false,
+		},
+	} {
+		t.Run(testcase.name, func(t *testing.T) {
+			sid, err := trace.SpanIDFromHex(testcase.hex)
+
+			if testcase.valid && err != nil {
+				t.Errorf("Expected SpanID %s to be valid but end with error %s", testcase.hex, err.Error())
+			}
+
+			if !testcase.valid && err == nil {
+				t.Errorf("Expected SpanID %s to be invalid but end no error", testcase.hex)
+			}
+
+			if sid != testcase.sid {
+				t.Errorf("Want: %v, but have: %v", testcase.sid, sid)
+			}
+		})
+	}
+}
+
+func TestMarshalTraceID(t *testing.T) {
+	for _, testcase := range []struct {
+		name string
+		tid  trace.ID
+		want string
+	}{
+		{
+			name: "Marshal TraceID",
+			tid:  trace.ID([16]byte{128, 241, 152, 238, 86, 52, 59, 168, 100, 254, 139, 42, 87, 211, 239, 247}),
+			want: `"80f198ee56343ba864fe8b2a57d3eff7"`,
+		},
+	} {
+		t.Run(testcase.name, func(t *testing.T) {
+			tid, err := json.Marshal(testcase.tid)
+			if err != nil {
+				t.Errorf("There is an error %s", err.Error())
+			} else if string(tid) != testcase.want {
+				t.Errorf("Want: %v, but have: %v", testcase.want, string(tid))
+			}
+		})
+	}
+}
+
+func TestUnmarshalTraceID(t *testing.T) {
+	for _, testcase := range []struct {
+		name string
+		data []byte
+		want trace.ID
+	}{
+		{
+			name: "Unmarshal TraceID",
+			data: []byte(`{"TraceID": "80f198ee56343ba864fe8b2a57d3eff7"}`),
+			want: trace.ID([16]byte{128, 241, 152, 238, 86, 52, 59, 168, 100, 254, 139, 42, 87, 211, 239, 247}),
+		},
+	} {
+		t.Run(testcase.name, func(t *testing.T) {
+			var tid trace.ID
```

Comment:
The data is in a struct format, the test should unmarshal into a similar struct format.

i.e.

```
type dump struct {
	ID ID `json:"some field name"`
}
d := new(dump)
err := json.Unmarshal(testcase.data, &d)
...
```